server/src/main/java/org/opensearch/monitor/fs/FsHealthService.java

```diff
@@ -134,6 +146,9 @@ public StatusInfo getHealth() {
             statusInfo = new StatusInfo(HEALTHY, "health check disabled");
         } else if (brokenLock) {
             statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
+        } else if (lastRunTimeMillis.get() > Long.MIN_VALUE && currentTimeMillisSupplier.getAsLong() -
+            lastRunTimeMillis.get() > refreshInterval.millis() + healthyTimeoutThreshold.millis()) {
+            statusInfo = new StatusInfo(UNHEALTHY, "healthy threshold breached");
```

Comment:
Can we simplify this condition by having current run start time, instead of relying on the last run time?

Reply:
Not sure how that would simplify